src/linked_list.py

```diff
@@ -47,39 +47,36 @@ def pop(self):
             self.header = pop_node.pointer
             self.length -= 1
             return pop_node.value
+        else:
+            raise IndexError('Cannot pop from an empty list.')
 
     def display(self):
         """Return a unicode string representing the list."""
         if self.length:
+            return_str = '{0}{1}{2}'.format('(', self.header.value, ', ')
             current_node = self.header
-            return_str = '('
             while current_node.pointer:
-                my_value = current_node.value
+                my_value = current_node.pointer.value
                 return_str += '{0}{1}'.format(my_value, ', ')
                 current_node = current_node.pointer
             else:
-                my_value = current_node.value
-                return_str += u'{0}{1}'.format(my_value, ')')
+                return_str = return_str.rstrip(', ') + ')'
             return return_str
 
     def search(self, val):
         """Return the node containing a 'val' in the list."""
         if self.length:
+            if self.header.value == val:
+                return self.header
+
             current_node = self.header
             while current_node.pointer:
-                if current_node.value == val:
-                    return current_node
+                if current_node.pointer.value == val:
+                    return current_node.pointer
                 current_node = current_node.pointer
```

Comment:
This method should give me a message telling me that my value is not in the list.